### DISCUSSION TITLE
Show More Discounting

### DISCUSSION THREAD

**Discussion by erikwest** [EvgeniyMukhamedjanov/liquid-ajax-cart#79] [Ideas]
Would be nice to show cart subtotal and total savings based on the product compare prices in cart. 

Ideally there is an code example of a cart subtotal & savings that can show complex discounting based on a cart with product compare prices + coupon codes + automatic discounts. Maybe even a conf that selects total savings based on price or compare price for specific use cases.

Also would be great to see a cart line item example with buy x get y automatic discounts applied.

Great project! Hope that helps.

**Comment by EvgeniyMukhamedjanov** [comment]
Hey, thank you for the feedback!
Do you mean it would be great to have more Shopify liquid examples how to show discounts in the cart?
Or you mean the demo store to have discounts for demonstration?

**Reply by erikwest** [reply]
Thanks for responding! 
Both would be great but liquid examples of a cart with a lot of discounting going on is all I really need personally.

**Reply by EvgeniyMukhamedjanov** [reply]
I see. 

Shopify has some kind of a "right" way of showing the discounts and the Dawn theme is a good reference as well as the shopify docs: https://shopify.dev/docs/api/liquid/objects/line_item#line_item-line_level_discount_allocations

So it depends on the liquid objects provided by Shopify rather than Liquid Ajax Cart.

If you personally need to show a specific discount or compare-at price, let me know, I'll provide a correct example.

**Reply by erikwest** [reply]
Would be great to get your input or liquid code examples for the following:

1. total compare price = total sum of all compare prices for products in cart (add regular price if no compare price on a product)
3. total savings = total compare price - total regular price + (any coupon codes + any automatic discounts)
4. cart line item buy x get y = dawn usually splits out a cart line item for this and adds the name of the automatic discount to each line item associated with the automatic discount.

**Reply by erikwest** [reply]
Was able to figure out total compare price and total savings via some liquid code snippets I found online.

In case anyone else needs:

```
{% assign total_cart_compare_price = 0 %}  

{% for line_item in cart.items %}
    {% if line_item.variant.compare_at_price != line_item.price and line_item.variant.compare_at_price > 0 %}
      {% assign compare_price = line_item.variant.compare_at_price | times: line_item.quantity %} 
    {% else %}
        {% assign compare_price = line_item.price | times: line_item.quantity %} 
    {% endif %}
    {% assign total_cart_compare_price = total_cart_compare_price | plus: compare_price %}
{ % endfor %}

Subtotal: <s>{{ total_cart_compare_price | money }}</s>
Savings: -{{ total_cart_compare_price | minus: cart.total_price | money }}
Total: {{ cart.total_price | money }}
```
